test/components/views/right_panel/UserInfo-test.tsx

```diff
@@ -0,0 +1,182 @@
+import { MatrixClientPeg } from '../../../../src/MatrixClientPeg';
+import MatrixClientContext from '../../../../src/contexts/MatrixClientContext';
+import VerificationPanel from '../../../../src/components/views/right_panel/VerificationPanel';
+import EncryptionInfo from '../../../../src/components/views/right_panel/EncryptionInfo';
+
+const findByTestId = (component, id) => component.find(`[data-test-id="${id}"]`);
+
+jest.mock('../../../../src/utils/DMRoomMap', () => {
+    const mock = {
+        getUserIdForRoomId: jest.fn(),
+        getDMRoomsForUserId: jest.fn(),
+    };
+
+    return {
+        shared: jest.fn().mockReturnValue(mock),
+        sharedInstance: mock,
+    };
+});
+
+describe('<UserInfo />', () => {
+    const defaultUserId = '@test:test';
+    const defaultUser = new User(defaultUserId);
+
+    const defaultProps = {
+        user: defaultUser,
+        phase: RightPanelPhases.RoomMemberInfo,
+        onClose: jest.fn(),
+    };
+
+    const mockClient = {
+        getUser: jest.fn(),
+        isGuest: jest.fn().mockReturnValue(false),
+        isUserIgnored: jest.fn(),
+        isCryptoEnabled: jest.fn(),
+        getUserId: jest.fn(),
+        on: jest.fn(),
+        isSynapseAdministrator: jest.fn().mockResolvedValue(false),
+        isRoomEncrypted: jest.fn().mockReturnValue(false),
+        doesServerSupportUnstableFeature: jest.fn().mockReturnValue(false),
+        mxcUrlToHttp: jest.fn().mockReturnValue('mock-mxcUrlToHttp'),
+        removeListerner: jest.fn(),
+        currentState: {
+            on: jest.fn(),
+        },
+    };
+
+    const verificationRequest = {
+        pending: true, on: jest.fn(), phase: Phase.Ready,
+        channel: { transactionId: 1 },
+        otherPartySupportsMethod: jest.fn(),
+    } as unknown as VerificationRequest;
+
+    const getComponent = (props = {}) => mount(<UserInfo {...defaultProps} {...props} />, {
+        wrappingComponent: MatrixClientContext.Provider,
+        wrappingComponentProps: { value: mockClient },
+    });
+
+    beforeAll(() => {
+        jest.spyOn(MatrixClientPeg, 'get').mockReturnValue(mockClient);
+    });
+
+    beforeEach(() => {
+        mockClient.getUser.mockClear().mockResolvedValue(undefined);
+    });
+
+    it('closes on close button click', () => {
+        const onClose = jest.fn();
+        const component = getComponent({ onClose });
+        act(() => {
+            findByTestId(component, 'base-card-close-button').at(0).simulate('click');
+        });
+
+        expect(onClose).toHaveBeenCalled();
+    });
+
+    describe('without a room', () => {
+        it('does not render space header', () => {
+            const component = getComponent();
+            expect(findByTestId(component, 'space-header').length).toBeFalsy();
+        });
+
+        it('renders user info', () => {
+            const component = getComponent();
+            expect(component.find(BasicUserInfo).length).toBeTruthy();
```

Comment:
Can we use `"BasicUserInfo"` instead of having to export it?

Reply:
good idea, done